include/proxy-wasm/wasm.h

```diff
@@ -224,6 +233,10 @@ class WasmBase : public std::enable_shared_from_this<WasmBase> {
   WasmCallVoid<1> on_log_;
   WasmCallVoid<1> on_delete_;
 
+  // Capabilities which are allowed to be linked to the module. If this is empty, restriction
+  // is not enforced.
+  absl::flat_hash_set<std::string> allowed_capabilities_;
```

Comment:
I think it's necessary to persist it even after `initialize` to support the Wasm cloning. When a thread-local Wasm is created it needs `allowed_capabilities_` to still exist in the base Wasm so that it can make its own copy.

Reply:
Cloning only happens during initialization, i.e. base WasmVm is created on the main thread, and then immediately we create clones on each worker thread, but we can copy data to the worker threads in the dispatching closure, so we don't need to persist that information.

Having said that, this list is going to be tiny compared to things like bytecode (see #94), so it might be premature optimization.

Reply:
Right. Let’s raise an issue or at least put a note on #94 about this optimization after this being merged